### ISSUE TITLE
Incorrect error when quoted type annotation is an f-string

### ISSUE THREAD

**Issue by JelleZijlstra** [microsoft/pyright#7640]
Consider this code:

```python
def invalid_strings(
    p1: r"int",  # E
    p2: f"int",  # E
    p3: b"int",  # E
):
    pass
```

This [produces](https://pyright-play.net/?pythonVersion=3.13&strict=true&code=CYUwZgBAlgdgbgQwDZWAfQM4BcBOsDmGAFAFATkQAOAjAFwQ4BEsWjANOQMQQCiZFlAEz0wzGKw4RufClQDM9AEZiJXXiQCUtfuUoIMGEkA) several pyright errors, including this one for the f-string:

```
Tuple expression not allowed in type annotation
  Use tuple[T1, ..., Tn] to indicate a tuple type or Union[T1, T2] to indicate a union type  (reportInvalidTypeForm)
```

That seems wrong, since there is no tuple involved in the code. I assume pyright desugars it to a tuple somewhere, but that should not leak into the user-visible error message.

Pyright allows the raw string and byte string annotations. Possibly it shouldn't, but I don't feel strongly.

I was looking at this because you (@erictraut) brought up exotic kinds of string annotations in the discussion around python/typing#1693. My current conclusion is that there's enough diversity in how type checkers treat these edge cases that I'd rather leave this to a different discussion.

**Comment by erictraut** [comment]
This is included in pyright 1.1.358, which I just published.
